substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/HeapImpl.java

```diff
@@ -87,59 +87,51 @@
 
 import jdk.vm.ci.meta.MetaAccessProvider;
 import jdk.vm.ci.meta.ResolvedJavaType;
-//Checkstyle: stop
 import sun.management.Util;
 //Checkstyle: resume
 
 /** An implementation of a card remembered set generational heap. */
-public class HeapImpl extends Heap {
+public final class HeapImpl extends Heap {
     /** Synchronization means for notifying {@link #refPendingList} waiters without deadlocks. */
     private static final VMMutex REF_MUTEX = new VMMutex();
     private static final VMCondition REF_CONDITION = new VMCondition(REF_MUTEX);
 
     // Singleton instances, created during image generation.
-    private final YoungGeneration youngGeneration;
-    private final OldGeneration oldGeneration;
-    final HeapChunkProvider chunkProvider;
-    private final ObjectHeaderImpl objectHeaderImpl;
+    private final YoungGeneration youngGeneration = new YoungGeneration("YoungGeneration");
```

Comment:
I think insignificantly so because none of the field initializers depend on each other and the constructor contains only initialization code that depends on the `FeatureAccess` argument or on some logic.